packages/api-explorer/__tests__/Doc.test.jsx

```diff
@@ -113,12 +114,119 @@ describe('state.dirty', () => {
 
   test('should switch to true on form change', () => {
     const doc = shallow(<Doc {...props} />);
-    doc.instance().onChange({ a: 1 });
+    const schema = {type: 'body', schema: {type: 'object', additionalProperties: true}}
+    const formData = { body: {a: 1} }
+    const data = {
+      formData,
+      schema
+    }
+    const previousFormData = doc.state().formData
+    doc.instance().onChange(data);
 
     expect(doc.state('dirty')).toBe(true);
+    expect(doc.state('formData')).not.toEqual(previousFormData)
+    expect(doc.state('formData')).toEqual(formData)
+  });
+
+  test('should not switch to true on form change without schema', () => {
+    const doc = shallow(<Doc {...props} />);
+    const formData = { body: {a: 1} }
+    const data = {
+      formData,
+    }
+    const previousFormData = doc.state().formData
+    doc.instance().onChange(data);
+
+    expect(doc.state('dirty')).toBe(false);
+    expect(doc.state('formData')).toEqual(previousFormData)
+  });
+
+  test('should not switch to true on form change with unmatched schema type', () => {
+    const doc = shallow(<Doc {...props} />);
+    const schema = {type: 'params', schema: {type: 'object', additionalProperties: true}}
+    const formData = { body: {a: 1} }
+    const data = {
+      formData,
+      schema
+    }
+    const previousFormData = doc.state().formData
+    doc.instance().onChange(data);
+
+    expect(doc.state('dirty')).toBe(false);
+    expect(doc.state('formData')).toEqual(previousFormData)
   });
 });
 
+test('should updated formData correctly on Params onChange with schema', () => {
+  const schemaFromOnchange = {
+    "type": "body",
+    "label": "body Params",
+    "schema": {
+      "type":"object",
+      "properties":{
+         "foo": {"type": "string"}
+      },
+      "additionalProperties":false
+   }
+  }
+
+  const element = shallow(<Doc
+    {...props}
+    stripSlash
+    doc={{
+        title: 'Title',
+        slug: 'slug',
+        type: 'endpoint',
+        swagger: { path: '/v2/crud/{id}' },
+        api: { method: 'post' },
+        formData: { body: { id: '1' }, auth: { api_key: '' } },
+        onSubmit: () => {},
+      }}
+    oas={GET_OAS_SCHEMA(schemaFromOnchange.schema)}
+  />)
+
+  element.find(Params).prop('onChange')({formData: {body: { id: '1' }}, schema: schemaFromOnchange})
+  expect(element.state().formData).toEqual({ body: { id: '1' } })
+
+  element.find(Params).prop('onChange')({formData: {body: {}}, schema: schemaFromOnchange})
+  expect(element.state().formData).toEqual({ body: {} })
+})
+
+test('should updated formData correctly on Params onChange without schema', () => {
```

Comment:
The schema property is set, the only difference seems to be the test assertion `with` --> `without`

Reply:
you're right.
Updated test